src/v2/api/controllers/RoleController.ts

```diff
@@ -155,53 +244,106 @@ export class RoleController {
-    const grantsMap = this.roleMapper.getGrants(grants);
-    return { grants: grantsMap };
+    const grants = await this.roleService.getGrants(roleId, query);
+    const mappedGrants = this.roleMapper.getGrants(grants.data);
+    return {
+      grants: mappedGrants,
+      pagination: grants.pagination,
+    };
   }
 
   @ApiBearerAuth()
   @ApiOkResponse({
-    type: BaseRoleResponse,
+    type: GrantResponse,
   })
   @ApiUnauthorizedResponse({
     description: `\n
     UnauthorizedException:
       Unauthorized`,
   })
   @ApiForbiddenResponse({
+    description: `\n 
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidBodyException:
+      Permission cannot be empty
+      Set is not a boolean`,
+  })
+  @ApiParam({
+    name: 'roleId',
+    required: true,
+    description: 'Id of certain role',
+  })
+  @ApiEndpoint({
+    summary: 'Give grants to the role by id',
+    permissions: PERMISSION.ROLES_GRANTS_CREATE,
+  })
+  @Post('/:roleId/grants')
+  async createGrants (
+      @Body() body: CreateGrantsDTO,
+      @Param('roleId') roleId: string,
+  ) {
+    return this.roleService.createGrants(roleId, body.grants);
+  }
+
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    type: GrantResponse,
+  })
+  @ApiUnauthorizedResponse({
     description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n 
     NoPermissionException:
       You do not have permission to perform this action`,
   })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidBodyException:
+      Permission cannot be empty
+      Set is not a boolean
+      Weight must be a number
+      Weight cannot be empty`,
+  })
   @ApiParam({
     name: 'roleId',
     required: true,
-    description: 'Id of the role to delete it',
+    description: 'Id of certain role',
   })
   @ApiEndpoint({
-    summary: 'Delete the role by id',
-    permissions: PERMISSION.ROLES_$ROLEID_DELETE,
+    summary: 'Create grant for certain role',
+    permissions: PERMISSION.ROLES_$ROLEID_GRANT_CREATE,
   })
-  @Delete('/:roleId')
-  async delete (
-    @Param('roleId') roleId: string,
+  @Post('/:roleId/grant')
+  async createGrant (
+      @Body() body: CreateGrantDTO,
+      @Param('roleId') roleId: string,
   ) {
-    const role = await this.roleService.delete(roleId);
-    return this.roleMapper.delete(role);
+    const grant = await this.roleService.createGrant(roleId, body);
+    return this.grantMapper.getMappedGrant(grant);
   }
 
   @ApiBearerAuth()
   @ApiOkResponse({
-    type: BaseRoleResponse,
+    type: MappedGrant,
   })
   @ApiBadRequestResponse({
     description: `\n
+    InvalidGrantIdException:
+      Grant with such id is not found
+      
     InvalidBodyException:
-      Name is not an enum
-      Weight is not a number`,
+      set must be boolean,
+      weight must be a number`,
```

Comment:
`Set` should start with a capital letter
`Weight` should start with a capital letter